1-Draft/RFCNNNN-Test-Connection.md

```diff
@@ -0,0 +1,121 @@
+---
+RFC: XXXX
+Author: Joel Sallow
+Status: Draft
+Version: 1.0
+Area: Cmdlets
+Comments Due: 6/10/19
+Plan to Implement: Yes
+---
+
+# Proposed Changes to `Test-Connection` Cmdlet
+
+## Syntax
+
+```
+# All Sets
+Test-Connection [-TargetName] <string[]> [-IPv4] [-IPv6] [-ResolveDestination] [-TimeoutSeconds <int>] [-Quiet] [<CommonParameters>]
+
+# Set 1
+[-Ping] [-Count <int>] [-Delay <int>] [-Source <string>] [-MaxHops <int>] [-BufferSize <int>] [-DontFragment]
+
+# Set 2
+[-Ping] [-Continues] [-Delay <int>] [-Source <string>] [-MaxHops <int>]  [-BufferSize <int>] [-DontFragment] 
+
+# Set 3
+Test-Connection [-TargetName] <string[]> -Traceroute [-Source <string>] [-MaxHops <int>] 
+
+# Set 4
+Test-Connection [-TargetName] <string[]> -MTUSizeDetect
+
+# Set 5
+Test-Connection [-TargetName] <string[]> -TCPPort <int> [-Source <string>]
+```
+
+## Changes
+
+1. All current host/information output is relegated to the `-Verbose` stream.
+2. No progress bar per committee decision in [PowerShell#6768](https://github.com/PowerShell/PowerShell/issues/6768).
+3. All output occurs as soon as data is available, individual records per ping reply / trace hop.
+4. All presented data should be readily available as properties to the user.
+
+## Output Type Proposed Members
+
+### PingStatus
+
+```csharp
+class PingStatus
+{
+    string Source { get; }
+    IPAddress Address { get; }
+    string Destination { get; }
+    int Latency { get; }
+    int BufferSize { get; }
+    PingOptions Options { get; }
+    // Hidden in the formatter when not set; only populated & shown when using the -MtuSizeDetect switch
+    int MtuSize { get; } = -1
```

Comment:
I missed a fairly necessary property here... `Status` should also be here.

Given the aforementioned issue with packets used for a traceroute returning `IPStatus.TtlExpired` I'm tossing up whether it might make sense to define it kind of like this:

```csharp
public IPStatus Status
{
    get => Options.Ttl < 128 && _reply.Status == IPStatus.TtlExpired
        // Treat "TtlExpired" as "Success" when this is a traceroute hop as that is the expected outcome.
        ? IPStatus.Success
        : _reply.Status;
}
```